mlx/dtype.h

```diff
@@ -39,7 +39,7 @@ struct Dtype {
   };
 
   Val val;
-  const uint8_t size;
+  uint8_t size;
```

Comment:
Have to drop the constness here since we'd need the Dtype to be copy-assignable, which we are going to need in `validate_shape()` so that we have a `Dtype` [defined first and assigned later](https://github.com/ml-explore/mlx/blob/352c625bcd8ef96bb25f18e7d5e3188993a4f901/python/src/array.cpp#L214-L234) to get the final promoted type.

Reply:
Can you keep this const (see comment below).

Reply:
Can you keep this const? See comment below.